### PR TITLE
Add dataclass to store kymoline refinement parameters

### PR DESCRIPTION
**Pull request by rpauszek**
**Why this PR?**
We will need access to parameters from the refinement methods in the future (for instance, localization precision). This PR sets up small dataclasses to store the info and isolate it away from `KymoLine`. We will then expose only what we want through public properties/methods in `KymoLine`

This is pulled from #262 since that issue is on hold until we can validate the algorithm

### CODE REVIEW

File: lumicks/pylake/kymotracker/detail/loc_models.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        Spatial coordinates in physical units.
+    """
+
+    pixel_size: float
+    position: np.ndarray
+
+    @property
+    def pixel_coordinate(self):
+        """Return spatial coordinates in units of pixels."""
+        return self.position / self.pixel_size
```
Comment (JoepVanlier) [inline lumicks/pylake/kymotracker/detail/loc_models.py:27]: I'd prefer if the responsibility for positional transforms to reside in the image to reduce the risk of things going out of sync.

File: lumicks/pylake/kymotracker/kymotracker.py [hunk 2]
```diff
@@ -290,7 +291,7 @@ def refine_lines_gaussian(
         Number of pixels on either side of the estimated line to include in the optimization data.
     refine_missing_frames : bool
         Whether to estimate location for frames which were missed in initial peak finding.
-    overlap_strategy : {'ignore', 'skip'}
+    overlap_strategy : {'multiple', 'ignore', 'skip'}
```
[the remark this answers is not in the document]
Comment (rpauszek) [reply lumicks/pylake/kymotracker/kymotracker.py:294]: unfortunately I forgot about this for the `0.12.0` release, but I guess it's not the most horrible thing in the world. still fixing it here

File: lumicks/pylake/kymotracker/detail/loc_models.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+@dataclass(frozen=True)
+class LocalizationModel:
+    """Helper (base) class to hold refinement optimization parameters.
+
+    Parameters
+    ----------
+    pixel_size : float
+        Pixel size in microns
```
[the remark this answers is not in the document]
Comment (rpauszek) [reply lumicks/pylake/kymotracker/detail/loc_models.py]: good point, indeed it's better to leave the conversion in `KymoLine`

File: lumicks/pylake/kymotracker/kymotracker.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply lumicks/pylake/kymotracker/kymotracker.py:294]: Yeah, no worries.

**Pull request merged**